Log reporter

rclee opened the pull request:
This adds a method to generate a view controller with a WKWebView to display event profiling information for debugging purposes.

rclee committed:
Begin work on log reporter port

rclee committed:
Finish Port

Still needs to be tested.

rclee committed:
Add navagation bar to log view

rclee committed:
Add button to dismiss

reverts some previous changes
spinner is not currently working

rclee committed:
Add script message controller

This is to allow us to be notified when the JS has completely loaded and will draw our webpage. Using this notification we can stop animating the activity indicator at the appropriate time.

boundsj commented on MapboxMobileEvents/MMEEventLogReportViewController.m:
```diff
@@ -0,0 +1,53 @@
+    [_spinner setFrame:CGRectMake(self.view.frame.size.width/2 - _spinner.frame.size.width/2, self.view.frame.size.height/2, _spinner.frame.size.width, _spinner.frame.size.height)];
+    [self.view addSubview:_spinner];
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
```
What is the purpose of marshaling to the main thread here? `viewDidLoad` would never be called in any other thread context, right?

boundsj reviewed (changes requested):
Thanks! This feature will really help profile performance of this library.

In addition to the questions I asked in-line, I was wondering how the `readAndDisplayLogFileFromDate:viewController` method is intended to be invoked. I don't think `MMEEventLogger` is or ever should be public so it is not something a host SDK or application can use. I think that means we would need to add an API to `MMEEventsManager` and the host SDK/app would need to know that API exists and call it and it, in turn, would use `MMEEventLogger`. 

Injecting this profile/HTML view is going to be a bit awkward no matter what but I wonder about an internal, undocumented API that automatically finds the current, topmost view controller and presents using whatever that is. Alternatively, I suppose we could design an API that vends a web view and the host SDK/app can be responsible for adding that web view to the hierarchy.

rclee commented:
Great feedback! I'll look into adding these changes tonight. I figured the `MMEEventLogger` should probably remain private but since it was just for debugging purposes that we might be able to get away with it. In hindsight the design you described makes more sense. 

I'll look into implementing the API for finding the topmost view controller, that is certainly cleaner than the current implementation. Thanks!

rclee committed:
Move HTML to file

-Move HTML to a file and loading from there
-Some cleanup and styling
-Added new files for UI Navigation utilites
-Added script message handler to feed data into js

rclee committed:
Add displayLogFileFromDate: to header file

rclee committed:
Move logger.html to Resources

rclee committed:
Switch to html string constant

boundsj reviewed (approved):
Looks good to me!
